File: testes_automacao/Teste.java

public class Teste{
    public static void main(String... args){
        long tempoInicial = System.nanoTime();
        int aux = 0;
        for (int i =0; i<=1000; i++){
            aux += i;
        }
        long tempoFinal = System.nanoTime();
        System.out.print("Tempo execução incremento JAVA ");
        System.out.println(tempoFinal-tempoInicial);
        // >>> Tempo execução incremento JAVA 7746 ns


        tempoInicial = System.nanoTime();
        int i = 0;
        while ( i<=1000000000 ){
            i++;
        }
        tempoFinal = System.nanoTime();
        System.out.print("Tempo execução loop JAVA ");
        System.out.println(tempoFinal-tempoInicial);
        // >>> Tempo execução loop JAVA 245155406 ns

        /* CONCLUSÃO
         *  O que se conclui é que java é brutalemente mais veloz devido ao seu JIT
         * a JVM trabalha com just in time por isso os resultados em nano segundo
         * algo como 10^-9, ou seja, 6 zeros de diferença.
         */
    }
}

File: testes_automacao/teste.ts
console.log("Teste execução");

// console.time("meu time");
function incremento(): number{
    var aux: number = 0;
    for (var i=0; i<=1000000000; i++) {
        aux += i;
    }
    return aux;
}

function loop (){
    var aux: number = 0;
    while (aux <= 1000000000 ) {
        aux++;
    }
    console.log(aux);
}

inicio = console.time();
incremento()
fim = console.timeEnd();
// console.timeEnd("fim tempo");
console.log("Tempo total incremento JS =>");
// >>> default: 865.757ms

inicio = console.time();
loop();
fim = console.timeEnd();
// console.timeEnd("fim tempo");
console.log("Tempo total loop JS =>");
// >>> default: 657.211ms

// CONCLUSÃO
//  Comparando os resultados com python é possível ver que as técnicas de otimização
// fizeram o script .py ficar ligeiramente mais rápido, diferença de 1ms, entretanto
// e também que o node é sim realmente mais rápido do que python, porém ao rodar o
// script diretamente no browser a diferença de desempenho em favor do python é brutal.